internal/config/config.go

```diff
@@ -552,11 +567,11 @@ type Config struct {
 }
 
 func (config *Config) setDefaults() {
-	config.Global.setDefaults()
+	isGlobalRoutingConfigSpecified := config.Global.setDefaults()
```

Comment:
I'm a bit confused. Can't we check `config.Global.Routing.IsEnhancedRoutingControlEnabled()` to see if global config is specified?

Reply:
oh yeah... i see. I'm having a very difficult time tracking down what is getting set and where.

Reply:
Good catch on the edge case. I don't have any concrete suggestions at the moment, but your renaming suggestion sounds good. Let's also add a comment about `isGlobalRoutingConfigSpecified`.